### PR TITLE
Turn 007 into practice

### PR DESCRIPTION
**Pull request by MissJae**
@cuihtlauac can you check this?

### CODE REVIEW

File: practice/007/answer/impl.ml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    | Many of 'a node list
+
+ let flatten list =
+   let rec aux acc = function
+     | [] -> acc
+     | One x :: t -> aux (x :: acc) t
+     | Many l :: t -> aux (aux acc l) t
+   in
+   List.rev (aux [] list)
+
```
Comment (divyankachaudhari) [inline practice/007/answer/impl.ml]: Please include a newline at end of the file.

File: practice/007/work/impl.ml [hunk 2]
```diff
@@ -0,0 +1 @@
+let flatten = failwith "Not yet implemented"
```
Comment (divyankachaudhari) [inline practice/007/work/impl.ml:5]: 1. Include the type definition in this template. 
2. Include a placeholder for the function's argument.

Comment (MissJae) [reply practice/007/work/impl.ml:5]: Done. Thanks

File: practice/007/answer/impl.ml (hunk 1 above)
Comment (cuihtlauac) [inline practice/007/answer/impl.ml]:
```suggestion

```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/007/answer/impl.ml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    | One of 'a 
+    | Many of 'a node list
+
+ let flatten list =
+   let rec aux acc = function
+     | [] -> acc
+     | One x :: t -> aux (x :: acc) t
+     | Many l :: t -> aux (aux acc l) t
+   in
+   List.rev (aux [] list)
```
Comment (cuihtlauac) [inline practice/007/answer/impl.ml]:
```suggestion
   List.rev (aux [] list)

```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]
Thanks @MissJae, this is good

**Comment by MissJae** [comment]
> Thanks @MissJae, this is good
Thank you @cuihtlauac

**Pull request merged**